hcn/hcnendpoint_test.go

```diff
@@ -182,6 +182,78 @@ func TestEndpointNamespaceAttachDetach(t *testing.T) {
 	}
 }
 
+// setupL4ProxyPolicyTest creates an endpoint inside a new overlay network and
+// returns the request to be sent.
+func setupL4ProxyPolicyTest(t *testing.T) (*HostComputeNetwork, *HostComputeEndpoint, PolicyEndpointRequest) {
+	network, err := CreateTestOverlayNetwork()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	endpoint, err := HcnCreateTestEndpoint(network)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	policySetting := L4ProxyPolicySetting{
+		Port: "80",
+		FilterTuple: FiveTuple{
+			Protocols:       "6",
+			RemoteAddresses: "10.0.0.4",
+			Priority:        8,
+		},
+		ProxyType: ProxyTypeWFP,
+	}
+
+	policyJSON, err := json.Marshal(policySetting)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	endpointPolicy := EndpointPolicy{
+		Type:     L4Proxy,
+		Settings: policyJSON,
+	}
+
+	request := PolicyEndpointRequest{
+		Policies: []EndpointPolicy{endpointPolicy},
+	}
+
+	return network, endpoint, request
+}
+
+// tearDownL4ProxyPolicyTest deletes the endpoint and the network that were
+// created for the test.
+func tearDownL4ProxyPolicyTest(t *testing.T, network *HostComputeNetwork, endpoint *HostComputeEndpoint) {
+	err := endpoint.Delete()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = network.Delete()
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestAddL4ProxyPolicyOnEndpoint(t *testing.T) {
+	network, endpoint, request := setupL4ProxyPolicyTest(t)
+	err := endpoint.ApplyPolicy(RequestTypeAdd, request)
+	if err != nil {
+		t.Fatal(err)
+	}
+	tearDownL4ProxyPolicyTest(t, network, endpoint)
+}
+
+func TestUpdateL4ProxyPolicyOnEndpoint(t *testing.T) {
```

Comment:
I would omit a test for Update policy. One test is enough to prove that the shim is able to program hns.